File: src-tauri/core/src/models/workspace.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceMeta {
    pub title: String,
    pub author: String,
    pub language: String,
    pub version: String,
    pub created: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SummaryEntry {
    pub title: String,
    pub path: String,
    pub level: u32,
    #[serde(default)]
    pub is_missing: bool,
    #[serde(default)]
    pub children: Vec<SummaryEntry>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RepairAction {
    pub kind: String,
    pub detail: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceInfo {
    pub root_path: String,
    pub workspace_meta: WorkspaceMeta,
    pub summary: Vec<SummaryEntry>,
    #[serde(default)]
    pub repairs: Vec<RepairAction>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ChapterInfo {
    pub name: String,
    pub path: String,
    pub index_path: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PageInfo {
    pub name: String,
    pub path: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ChapterOrder {
    pub path: String,
    pub new_order: u32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MdFileContent {
    pub path: String,
    pub content: String,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AssetInfo {
    pub name: String,
    pub size: u64,
    pub file_type: String,
    pub path: String,
}

File: src/types/workspace.ts
export interface WorkspaceMeta {
  title: string;
  author: string;
  language: string;
  version: string;
  created: string;
}

export interface SummaryNode {
  title: string;
  path: string;
  level: number;
  isMissing?: boolean;
  children?: SummaryNode[];
}

export interface RepairAction {
  kind: 'added_missing_chapter' | 'missing_file' | 'field_defaulted';
  detail: string;
}

export interface WorkspaceInfo {
  rootPath: string;
  workspaceMeta: WorkspaceMeta;
  summary: SummaryNode[];
  repairs: RepairAction[];
}

export interface ChapterInfo {
  name: string;
  path: string;
  indexPath: string;
}

export interface PageInfo {
  name: string;
  path: string;
}

export interface ChapterOrder {
  path: string;
  newOrder: number;
}

export interface RecentWorkspace {
  path: string;
  title: string;
  lastOpened: string;
}

export interface FileTreeNode {
  name: string;
  path: string;
  type: 'folder' | 'file';
  children?: FileTreeNode[];
  isMissing?: boolean;
}

export interface WhiteboardAnchor {
  sourceFilePath: string;
  cursorPosition: number;
  nearestHeading: string;
}
